moment_kinetics/src/r_advection.jl

```diff
@@ -47,6 +48,43 @@ do a single stage time advance (potentially as part of a multi-stage RK scheme)
     end
 end
 
+"""
+1D false r advection, intended to provoke ITG in 1D. Despite there being no real r 
+advection, we insist that a radial temperature gradient exists, and add the ExB drift
+term to the DKE anyway to bring in temperature gradient drive. Since f is not known 
+along r, and hence its temperature dependence is not known, it is assumed that f 
+is roughly a Maxwellian with an R dependent temperature only. This means we can control
+L_T through this term.
+"""
+function r_advection_1D_ITG!(
+                         f_out, fvec_in, moments, fields, advect, r, z, vperp, vpa, dt,
+                         r_spectral, composition, geometry, scratch_dummy)
+
+    # only needs to be one of the terms because it's constant in z and r
+    B_factor = geometry.Bzeta[1,1] / geometry.Bmag[1,1]^2
+    L_T = composition.ion[1].L_T
+    #println("B_factor = $B_factor")
+    #println("L_T = $L_T")
+
+    T = zeros(size(moments.ion.dens))
+    @. T = moments.ion.p/moments.ion.dens
+    f = fvec_in.pdf
+    Ez = fields.gEz
+
+    # Expression is B_zeta/B^2 * Ez * f/T * dT/dr * (m(vperp^2 + vpa^2)/2T - 3/2)
+    for is in 1:1
+        for iz in 1:z.n
+            for ivperp in 1:vperp.n
+                for ivpa in 1:vpa.n
+                    f_out[ivpa,ivperp,iz,1,is] += B_factor * Ez[ivperp,iz,1,is] * f[ivpa,ivperp,iz,1,is]/L_T *
+                                                    (1/2 * (vperp.grid[ivperp]^2 + vpa.grid[ivpa]^2)/T[iz,1,is] - 3/2) * dt
+                end
+            end
+        end
+    end

```

Comment:
This will only work in serial. Could wrap in a `@begin_serial_region(); @serial_region begin...end`, but probably better/easier to just use the `@loop_*` macro for the loop. If you move the calculation of `T` inside the species and z loops, and just calculate it as a scalar variable for that `is` and `iz`, then there would be no need to allocate/fill an array for `T`.